app/gui/view/execution-mode-selector/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+#![allow(clippy::let_and_return)]
+// === Non-Standard Linter Configuration ===
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use enso_prelude::*;
+use ensogl::prelude::*;
+
+use enso_frp as frp;
+use ensogl::application::Application;
+use ensogl::data::color::Rgba;
+use ensogl::display;
+use ensogl::display::camera::Camera2d;
+use ensogl_gui_component::component;
+use ensogl_hardcoded_theme::graph_editor::execution_mode_selector as theme;
+
+
+
+// ================
+// === Constants ===
+// ================
+
+const PLAY_BUTTON_SIZE: f32 = 10.0;
+const PLAY_BUTTON_OFFSET: f32 = 15.0;
+const PLAY_BUTTON_PADDING: f32 = 10.0;
+const DIVIDER_OFFSET: f32 = 32.5;
+const DIVIDER_PADDING: f32 = 10.0;
+const DROPDOWN_WIDTH: f32 = 95.0;
+/// Width of the execution mode selector.
+pub const OVERALL_WIDTH: f32 =
+    DROPDOWN_WIDTH + 2.0 * DIVIDER_PADDING + PLAY_BUTTON_SIZE + PLAY_BUTTON_PADDING;
+/// Height of the execution mode selector.
+pub const HEIGHT: f32 = 24.0;
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+mod background {
+    use super::*;
+
+    ensogl::shape! {
+        below = [play_icon];
+        (style:Style) {
+            let size = Var::canvas_size();
+            let height = size.y();
+            let width = size.x();
+            let color = style.get_color(theme::background);
+            let base_rect = Rect(size).corners_radius(&height/2.px()).fill(color);
+
+            let divider = Rect((1.px(), &height)).fill(style.get_color(theme::divider));
+            let divider = divider.translate_x(width/2.0 - DIVIDER_OFFSET.px());
+
+            (base_rect + divider).into()
+        }
+    }
+}
+
+mod play_icon {
+    use super::*;
+
+    use std::f32::consts::PI;
+    ensogl::shape! {
+        above = [background];
+        (style:Style) {
+           let size = Var::canvas_size();
+           let height = size.y();
+           let width = size.x();
+           let triangle = Triangle(width, height).rotate((PI/2.0).radians());
+           let color = style.get_color(theme::triangle);
+            triangle.fill(color).into()
+        }
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+/// An identifier of a execution mode.
+pub type ExecutionMode = String;
+/// A list of execution modes.
+pub type ExecutionModes = Rc<Vec<ExecutionMode>>;
+
+ensogl::define_endpoints_2! {
+    Input {
+        set_execution_modes      (ExecutionModes),
+    }
+    Output {
+        selected_ecxecution_mode (ExecutionMode),
+        play_press(),
+    }
+}
+
+
+// =============
+// === Model ===
+// =============
+
+/// The model of the execution mode selector.
+#[derive(Debug, Clone, CloneRef)]
+pub struct Model {
+    display_object: display::object::Instance,
+    inner_root:     display::object::Instance,
```

Comment:
no description of what it is. Either docs or a better name would be appreciated here.